go/libraries/doltcore/sql/sqltestutil/insertqueries.go

```diff
@@ -0,0 +1,126 @@
+		ExpectedSchema: CompressSchema(PeopleTestSchema),
+	},
+	{
+		Name:           "insert full columns mixed order",
+		InsertQuery:    "insert into people (num_episodes, uuid, rating, age, is_married, last, first, id) values (222, '00000000-0000-0000-0000-000000000002', 9, 10, false, 'Simpson', 'Bart', 2)",
+		SelectQuery:    "select * from people where id = 2",
+		ExpectedRows:   CompressRows(PeopleTestSchema, Bart),
+		ExpectedSchema: CompressSchema(PeopleTestSchema),
+	},
+	{
+		Name:           "insert partial columns",
+		InsertQuery:    "insert into people (id, first, last) values (2, 'Bart', 'Simpson')",
+		SelectQuery:    "select id, first, last from people where id = 2",
+		ExpectedRows:   Rs(NewResultSetRow(types.Int(2), types.String("Bart"), types.String("Simpson"))),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind),
+	},
+	{
+		Name:           "insert partial columns mixed order",
+		InsertQuery:    "insert into people (last, first, id) values ('Simpson', 'Bart', 2)",
+		SelectQuery:    "select id, first, last from people where id = 2",
+		ExpectedRows:   Rs(NewResultSetRow(types.Int(2), types.String("Bart"), types.String("Simpson"))),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind),
+	},
+	{
+		Name:        "insert missing non-nullable column",
+		InsertQuery: "insert into people (id, first) values (2, 'Bart')",
+		ExpectedErr: "column <last> received nil but is non-nullable",
+	},
+	{
+		Name:        "insert partial columns mismatch too many values",
+		InsertQuery: "insert into people (id, first, last) values (2, 'Bart', 'Simpson', false)",
+		ExpectedErr: "too many values",
+	},
+	{
+		Name:        "insert partial columns mismatch too few values",
+		InsertQuery: "insert into people (id, first, last) values (2, 'Bart')",
+		ExpectedErr: "too few values",
+	},
+	{
+		Name:           "insert partial columns functions",
+		InsertQuery:    "insert into people (id, first, last) values (2, UPPER('Bart'), 'Simpson')",
+		SelectQuery:    "select id, first, last from people where id = 2",
+		ExpectedRows:   Rs(NewResultSetRow(types.Int(2), types.String("BART"), types.String("Simpson"))),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind),
+	},
+	{
+		Name:           "insert partial columns multiple values",
+		InsertQuery:    "insert into people (id, first, last) values (0, 'Bart', 'Simpson'), (1, 'Homer', 'Simpson')",
+		SelectQuery:    "select id, first, last from people where id < 2 order by id",
+		ExpectedRows:   Rs(NewResultSetRow(types.Int(0), types.String("Bart"), types.String("Simpson")),
+			NewResultSetRow(types.Int(1), types.String("Homer"), types.String("Simpson"))),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind),
+	},
+	{
+		Name:        "insert partial columns existing pk",
```

Comment:
Should also have a test that uses AdditionalSetupFn to set up a database that already has the row being duplicated before the insert statement.